codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala

```diff
@@ -176,6 +160,69 @@ class SerializerCheckCompilerPluginComponent(
             None
         }
       }
+
+      private def collectMissingClassNames(fullyQualifiedClassNames: List[String]): List[String] = {
```

Comment:
If I may provide an alternative solution to this problem.

For this problem I originally thought about reusing already existing compiler phase.

The non-existing FQCN are resolved by typer compiler phase (you can check this by using `-Ystatistics`)

So if we could create our own typer phase (by `global.analyzer.newTyper`) and send it prepared by plugin AST that contains the missing FQCN the phase would resolve these names for us.

Reply:
Ok, can you specify why adding our own custom typer phase would be a better solution?

Reply:
My first thought was - it would be smart but it also adds one more phase to each compilation - whereas my solution adds additional method invocation only in situations, where there were some missing types found. In my opinion we do not need one more phase for this particular reason (these false positives should not be that often? rather a rare case?)

Reply:
> method invocation only in situations, where there were some missing types found

This doesn't change regardless of method

Reply:
> why adding our own custom typer phase would be a better solution?

It doens't use fat jar, better files and also it does not depend on sbt.

Reply:
(@LukaszKontowski consider opening a separate _self-contained_ issue, i.e. with all known details outlined &mdash; no reference to this thread, for the sake of clarity)

Reply:
It's hard to me to assess how significant the suggested improvement is... but I'd intuitively go for 2. as well 😅

Reply:
Sure, I'll create a new issue for this improvement with detailed description.